examples/groth16_multigarble_181.rs

```diff
@@ -0,0 +1,155 @@
+    pub a: Option<F>,
+    pub b: Option<F>,
+    pub num_variables: usize,
+    pub num_constraints: usize,
+}
+
+impl<F: ark::PrimeField> ark::ConstraintSynthesizer<F> for DummyCircuit<F> {
+    fn generate_constraints(
+        self,
+        cs: ark::ConstraintSystemRef<F>,
+    ) -> Result<(), ark::SynthesisError> {
+        let a = cs.new_witness_variable(|| self.a.ok_or(ark::SynthesisError::AssignmentMissing))?;
+        let b = cs.new_witness_variable(|| self.b.ok_or(ark::SynthesisError::AssignmentMissing))?;
+        let c = cs.new_input_variable(|| {
+            let a = self.a.ok_or(ark::SynthesisError::AssignmentMissing)?;
+            let b = self.b.ok_or(ark::SynthesisError::AssignmentMissing)?;
+            Ok(a * b)
+        })?;
+
+        for _ in 0..(self.num_variables - 3) {
+            let _ =
+                cs.new_witness_variable(|| self.a.ok_or(ark::SynthesisError::AssignmentMissing))?;
+        }
+
+        for _ in 0..self.num_constraints - 1 {
+            cs.enforce_constraint(ark::lc!() + a, ark::lc!() + b, ark::lc!() + c)?;
+        }
+
+        cs.enforce_constraint(ark::lc!(), ark::lc!(), ark::lc!())?;
+        Ok(())
+    }
+}
+
+/// Run a batch of N garbled circuit verifications in parallel.
+///
+/// This function creates N garbling lanes, each with its own seed,
+/// and runs them simultaneously using MultigarblingMode. The AES accumulating
+/// hash batch collects ciphertexts from all lanes.
+///
+fn run_batch<const N: usize>(
+    inputs: &garbled_groth16::GarblerInput,
+    cap: usize,
+    seeds: &[u64],
+) -> Vec<[u8; 16]> {
+    let mut seeds_array: [u64; N] = [0; N];
+    let len = seeds.len();
+    for (j, &s) in seeds.iter().enumerate() {
+        seeds_array[j] = s;
+    }
+    let last_seed = seeds[len - 1];
+    for item in seeds_array.iter_mut().skip(len) {
+        *item = last_seed;
+    }
+
+    CircuitBuilder::run_streaming::<_, _, Vec<_>>(
+        inputs.clone(),
+        garbled_snark_verifier::circuit::modes::MultigarblingMode::<
+            AesNiHasher,
+            AESAccumulatingHashBatch<N>,
+            N,
+        >::new(cap, seeds_array, AESAccumulatingHashBatch::<N>::default()),
+        |root, input| vec![garbled_groth16::verify(root, input)],
+    )
+    .ciphertext_handler_result
+    .into_iter()
+    .take(len)
+    .collect::<Vec<[u8; 16]>>()
+}
+
+fn main() {
+    garbled_snark_verifier::init_tracing();
+
+    let cap = 150_000;
+    let mut rng = ChaCha20Rng::seed_from_u64(12345);
+    let circuit = DummyCircuit::<ark::Fr> {
+        a: Some(ark::Fr::rand(&mut rng)),
+        b: Some(ark::Fr::rand(&mut rng)),
+        num_variables: 10,
+        num_constraints: 1 << 10,
+    };
+    let (_pk, vk) = ark::Groth16::<ark::Bn254>::setup(circuit, &mut rng).expect("setup");
+
+    let inputs = garbled_groth16::GarblerInput {
+        public_params_len: 1,
+        vk: vk.clone(),
+    };
+
+    // Number of parallel garbling lanes (batch size).
+    const N: usize = 8;
```

Comment:
Change the name and move it to the beginning. Also replace 181 in the name with `multithread_multigarbling` and have 181 constant at the beginning